File: C++/binarysearch.cpp
// Binary Search Algorithm in C++

#include <iostream>

using namespace std;

int BinarySearch(int arr[], int size, int x)
{
    
    int s=0;
    int e=size-1;
    //2 pointer
    while(s<=e){
        int mid=(s+e)/2;
        if(arr[mid]==x){
            return mid;
        }else if(arr[mid]<x){
            s=mid+1;
            return s;
        }else{
            e=mid-1;
            return e;
        }
    }

    return -1;

}

int main()
{
    int arr[5] = {2, 3, 4, 10, 40};
    int x = 10;
    int z = BinarySearch(arr, 5, x);
    cout << z;
    return 0;
}

File: Java/BinarySearch.java
// Binary Search Algorithm in Java

public class BinarySearch {
    // Array of integers
    static int[] arr = { 2, 3, 4, 10, 40 };
    // Element to be searched
    static int x = 10;

    public static void main(String[] args) {
        // Binary Search
        
        System.out.println(binarySearch(arr, x));
    }

    static int binarySearch(int[] arr, int x) {
        // Binary Search
        // your code here
        int first = 0;
        int last = arr.length-1;
        if ((last-first)>1){
            int mid = (last+first)/2;
            if (arr[mid]<x){
                first = mid+1;
            }
            else{
                last=mid;
            }
        }
        if (arr[first]==x){
            return first;
        }
        else if (arr[last]==x){
            return last;
        }
        else{
            return -1;
        }

    }

}

// Output:
// 3
